### ISSUE TITLE
Using variables with custom embedded argument regular expressions

### ISSUE THREAD

**Issue by tester-512** [nokia/RED#314]
According to the User Guide "Robot Framework automatically enhances the specified regexps so that they match variables in addition to the text matching the pattern. This means that it is always possible to use variables with keywords having embedded arguments." In RED however, it says Unknown Keyword if you put a variable in the keyword name. Take this example
```
*** Test Cases ***
Test
    Do Stuff With Tiger

*** Keywords ***

Do Stuff With ${Animal:(Cat|Tiger)}
    ${Animal} Walk

${Animal:(Dog|Cat|Tiger)} Walk
    Log    ${Animal} Walked
```

In the above example, it logs Tiger Walked. In RED however, the line ${Animal} Walk would say that the keyword is unknown, yet in the RF user guide it says that it ignores the regex of the keyword being called if there is a variable in the name. RED shouldn't say that the keyword is unknown. Instead, it should only caution users that it cannot verify that the keyword exists. 

IF POSSIBLE:
Or, if it is possible, it would be beneficial to check the regex of the given variable and see if it matches (or is encapsulated by) the regex of the keyword being called, however that may run into problems if there are multiple keywords using the same format but with different regexs. Take this example:

```
Do Stuff With ${Animal:(Cat|Tiger)}
    ${Animal} Walk

${Animal:(Dog)} Walk
    Log    Dog:${Animal} Walked

${Animal:(Tiger)} Walk
    Log    Tiger:${Animal} Walked
```

RF errors out saying that there are multiple keywords matching name '${Animal} Walk' because in RF's view, it completely ignores the regex. In this case, ${Animal} Walk should show an error in RED.

In this next example, it should not show an error, but instead warn the user that the regex does not match:
```
Do Stuff With ${Animal:(Cat|Tiger)}
    ${Animal} Walk

${Animal:(Dog)} Walk
    Log    Dog:${Animal} Walked
```

This runs perfectly fine in RF.

**Comment by michalanglart** [comment]
Hi,

I've investigated the issue and here are the findings:

1. We can fix the keyword searching mechanism so that there will be no error in examples 1 and 3 and "ambigous keyword" error in case of 2 example
2. We may issue a warning in case of parameterized calls (calls like `${Animal} Walk`). There is actually already a similar warning raised in case of Test/Suite Setup/Teardown settings.
3. I don't think we can statically analyze the 3rd example in order to tell that the regexes used there do not match. It is interesting however that RF runs fine in this example because in runtime it should see that `Tiger` is passed so I would expect it to fail and tell that there is no keyword which would match `Tiger Walk`

**Comment by michalanglart** [comment]
Fixed in 0.9.0
